gui/src/components/Balances.tsx

```diff
@@ -18,22 +18,30 @@ export function Balances() {
     "db_get_erc20_balances",
     { address }
   );
+  const { data: settings } = useInvoke<GeneralSettings>("settings_get");
 
   useRefreshTransactions(mutate);
 
   return (
     <Panel>
       <Stack>
         {address && <BalanceETH address={address} />}
-        {(balances || []).map(([contract, balance]) => (
-          <BalanceERC20
-            key={contract}
-            {...{
-              contract,
-              balance: BigNumber.from(balance),
-            }}
-          />
-        ))}
+        {(balances || [])
+          .filter(([, balance]) =>
+            settings?.hideEmptyTokens ? !BigNumber.from(balance).isZero() : true
```

Comment:
I moved to viem in a recent PR, so you'll need to change this to use `BigInt` instead of `BigNumber`